Docs

brunozc opened the pull request.

brunozc committed:
draft documentation

brunozc committed:
Merge branch 'main' into docs

brunozc committed:
added first draft of nightly_build

brunozc committed:
Update documentation for tutorials and API definitions

brunozc committed:
Add introduction chapter and update benchmark report structure

- Introduced a new introduction chapter providing an overview of the STEM software and its validation process.
- Enhanced the main report structure with improved title page, header, and footer settings.
- Updated output settings in 2D and 3D strip load scripts for finer output control.

brunozc committed:
Add nightly benchmark workflow and initial report template

brunozc committed:
Update documentation and improve benchmark configurations

- Refine cron schedule comment in nightly benchmark workflow.
- Enhance formatting in introduction chapter and other LaTeX files.
- Add missing reference for Newmark's method in bibliography.
- Adjust mesh size parameter in one-dimensional wave propagation script.
- Update dynamic strip load documentation for clarity and consistency.
- Improve dynamic strip load on 3D elastic halfspace documentation.

brunozc committed:
Refactor sys.path handling in runner.py for improved script execution

brunozc committed:
Merge branch 'main' into docs

brunozc committed:
updated build

brunozc committed:
Upgrade pip before installing Python dependencies in nightly build workflow

brunozc committed:
Add installation step for libGLU library in nightly build workflow

brunozc committed:
Refactor comments in runner.py for clarity and consistency

brunozc committed:
Merge branch 'main' into docs

brunozc committed:
Fix comment to accurately describe coordinates for the 2D block in test_stem function

brunozc committed:
Add moving load benchmark documentation and reference for Fryba 2013

brunozc committed:
Refactor: Replace LinearNewtonRaphsonStrategy with NewtonRaphsonStrategy in strip_load_2D.py and strip_load_3D.py

brunozc committed:
Remove tqdm progress bar from file reading loop in read_VTK.py

brunozc committed:
Add Lamb problem benchmark documentation and associated figures

brunozc committed:
Add draft moving load analytical solution and update documentation

brunozc committed:
WIP moving load

brunozc committed:
Refactor: Update Pekeris runner parameters and remove unused imports

brunozc committed:
Refactor: Improve moving load calculations and update output settings in strip load runners. added SDOF

brunozc committed:
Update benchmark report and related files

- Adjust watermark scale in benchmark report to improve visibility.
- Reorganize input order in benchmark report, including new SDOF section and replacing Pekeris with Lamb.
- Modify Lamb section to correct soil dimensions and update figure references from PDF to PNG.
- Remove obsolete mesh PDF file for Lamb.
- Update SDOF figure size for better presentation.
- Refactor compare_analytical.py to include commented-out code for moving load analysis.
- Update runner.py to streamline execution of various analyses and remove commented-out moving load code.
- Correct element sizes in strip load 2D and 3D sections for accuracy.
- Update mesh images for strip load 2D and 3D to reflect recent changes.

brunozc committed:
added missing file

brunozc committed:
Add one-dimensional wave propagation benchmark documentation

brunozc committed:
Update LaTeX compilation method and font settings in benchmark report

brunozc committed:
Merge branch 'main' into docs

brunozc committed:
Update nightly build to install additional LaTeX packages

Added installation of texlive-xetex and fonts-noto for LaTeX.

brunozc committed:
Merge branch 'docs' of github.com:StemVibrations/STEM into docs

brunozc committed:
added missing file

brunozc committed:
fixed typo

brunozc committed:
corrected fonts

brunozc committed:
fixed font

aronnoordam committed:
added vibrating dam section to verification docs

aronnoordam committed:
added vibrating dam.tex

aronnoordam committed:
added vibrating dam 3d docs

aronnoordam committed:
precommit

aronnoordam committed:
precommit

aronnoordam committed:
Merge remote-tracking branch 'refs/remotes/origin/bousinesq_3d' into docs_ver/bousinesq

aronnoordam committed:
Update nightly_build/vibrating_dam_3D/vibrating_dam_3d.tex

Co-authored-by: brunozc <brunozc@users.noreply.github.com>

aronnoordam committed:
Update nightly_build/vibrating_dam/vibrating_dam.tex

Co-authored-by: brunozc <brunozc@users.noreply.github.com>

aronnoordam committed:
Merge pull request #348 from StemVibrations/docs_ver/add_vibrating_dam_doc

Docs ver/add vibrating dam doc

aronnoordam committed:
added boussinesq comparison for docs

aronnoordam committed:
Merge branch 'docs' into docs_ver/bousinesq

# Conflicts:
#	nightly_build/benchmark_report.tex
#	nightly_build/references.bib
#	nightly_build/stem_runners/compare_analytical.py
#	nightly_build/stem_runners/runner.py

aronnoordam committed:
added uniformly loaded simply supported beam doc

aronnoordam committed:
precommit

aronnoordam committed:
added 1d abs boundary verification doc

aronnoordam committed:
added 2d case for 1d abs

aronnoordam committed:
applied comments

aronnoordam committed:
corrected tex

aronnoordam committed:
Merge pull request #353 from StemVibrations/docs_ver/simply_supported_beam

Docs ver/simply supported beam

aronnoordam committed:
Merge branch 'docs' into docs_ver/1d_abs

# Conflicts:
#	nightly_build/stem_runners/compare_analytical.py
#	nightly_build/stem_runners/runner.py

aronnoordam committed:
merge

aronnoordam committed:
Merge pull request #352 from StemVibrations/docs_ver/1d_abs

docs_ver/1d_abs

brunozc committed:
updated tex

brunozc committed:
updated runner

brunozc committed:
Merge branch 'main' into docs

brunozc committed:
updated ref

aronnoordam committed:
added moving load on beam tex

aronnoordam committed:
precommit

aronnoordam committed:
Merge branch 'docs' into docs_ver/moving_load_on_beam

# Conflicts:
#	nightly_build/stem_runners/compare_analytical.py

aronnoordam committed:
clarification

aronnoordam committed:
Merge branch 'docs' into docs_ver/bousinesq

# Conflicts:
#	nightly_build/stem_runners/compare_analytical.py
#	nightly_build/stem_runners/runner.py

aronnoordam committed:
Merge branch 'main' into docs_ver/bousinesq

# Conflicts:
#	benchmark_tests/analytical_solutions/boussinesq.py

aronnoordam committed:
cleanup

aronnoordam committed:
Merge branch 'docs' into docs_ver/moving_load_on_beam

# Conflicts:
#	docs/refs.bib

brunozc committed:
Add dynamic strip load benchmark for elastic halfspace

- Introduced a new chapter on dynamic strip load in a 2D elastic halfspace, including model description, material parameters, and results.
- Removed outdated 3D strip load benchmark files and associated figures.
- Updated references to ensure consistency with the new 2D benchmark.

aronnoordam committed:
Merge branch 'docs' into docs_ver/moving_load_on_beam

# Conflicts:
#	nightly_build/stem_runners/runner.py

aronnoordam committed:
merge

aronnoordam committed:
Merge branch 'docs' into docs_ver/bousinesq

# Conflicts:
#	nightly_build/benchmark_report.tex
#	nightly_build/stem_runners/runner.py

aronnoordam committed:
merge

aronnoordam committed:
uncommited comparison

aronnoordam committed:
style

aronnoordam committed:
precommit

aronnoordam committed:
style

aronnoordam committed:
applied comments

aronnoordam committed:
Update nightly_build/moving_load_on_beam/moving_load_on_beam.tex

Co-authored-by: brunozc <brunozc@users.noreply.github.com>

aronnoordam committed:
Merge pull request #354 from StemVibrations/docs_ver/bousinesq

Docs ver/bousinesq

aronnoordam committed:
Merge branch 'docs' into docs_ver/moving_load_on_beam

# Conflicts:
#	nightly_build/stem_runners/runner.py

aronnoordam committed:
Merge pull request #356 from StemVibrations/docs_ver/moving_load_on_beam

Docs ver/moving load on beam

brunozc committed:
updated docs

brunozc committed:
Merge branch 'docs' of github.com:StemVibrations/STEM into docs

brunozc committed:
fix ref

brunozc committed:
added missing file

brunozc committed:
updated docs

brunozc committed:
Update material properties and test configurations across multiple files

- Changed solid density from 2700 kg/m³ to 2000 kg/m³ and updated porosity to 0 in one-dimensional boundary and wave propagation tests.
- Enhanced the introduction section with total counts of unit and benchmark tests implemented.
- Modified the comparison script for analytical solutions to include additional nodes and improved plotting for clarity.
- Adjusted the runner scripts to reflect updated material properties and ensure consistency across different test scenarios.
- Cleaned up commented-out code and improved plotting aesthetics in the comparison functions.

brunozc committed:
Refactor documentation references in LaTeX files for consistency

brunozc committed:
Add moving load benchmark for elastic halfspace analysis

- Introduced a new LaTeX document detailing the moving load on an elastic halfspace, including model description, materials, numerical parameters, and results.
- Updated the Python script for comparing analytical and numerical results to include vertical displacement calculations for the moving load scenario.
- Modified the moving load halfspace runner to adjust boundary conditions and load parameters for improved accuracy in simulations.

brunozc committed:
Enhance documentation on boundary conditions, formulations, and materials with detailed examples and clarifications

brunozc committed:
Enhance documentation for load definitions with detailed examples and clarifications

brunozc committed:
Enhance documentation on boundary conditions, outputs, and configuration settings with additional examples and clarifications

brunozc committed:
Add Rayleigh damping illustration and enhance solver settings documentation

brunozc committed:
Refactor documentation for authors, boundary conditions, loads, and index; remove obsolete stem package documentation

brunozc committed:
Add tutorial for UVEC model with rail joint and static initialization

This commit introduces a comprehensive tutorial on setting up a train model using the User defined VEhiCle (UVEC) model. The tutorial covers the entire process, including model initialization, soil material specification, geometry definition, track generation, rail joint modeling, UVEC load application, boundary conditions, solver settings, and output configuration. The tutorial is designed to guide users through both static initialization and dynamic analysis stages of the model.

brunozc committed:
Merge branch 'main' into docs

brunozc committed:
Refactor displacement constraint parameters in UVEC tutorial for clarity

brunozc committed:
Refactor tutorial documentation and remove obsolete VTK reading script

- Updated tutorial4.rst, tutorial5.rst, and tutorial6.rst to replace "# END CODE BLOCK" comments with ".. # END CODE BLOCK" for consistency in documentation style.
- Removed the read_VTK.py script from the nightly build stem runners as it is no longer needed.

brunozc committed:
Improve clarity and consistency in tutorial documentation for Lamb's problem and stationary load examples

brunozc committed:
Add visual assets and update documentation for Lamb's problem and line load tutorial

brunozc committed:
Remove tutorial6.rst: This file provided a detailed guide on setting up a train model (UVEC) with rail joint and static initialization. The content included model setup, soil material specifications, rail track generation, UVEC model definition, boundary conditions, solver settings, and output configurations. The removal is part of a restructuring of the documentation.

brunozc committed:
Delete test_tutorial_6 function

Removed test function for tutorial 6 from benchmark tests.

aronnoordam commented on nightly_build/lamb/mesh.svg:
also the coordiantes are not the same as in the pyhton file or as in the pdf

brunozc committed:
Add API reference documentation and update various tutorial sections for clarity

brunozc committed:
Refactor tutorial3.rst to enhance clarity and update model geometry

- Updated the tutorial to describe a 3D geometry with an embankment and bridge.
- Revised imports and setup section to reflect new input folder structure.
- Clarified the geometry and materials section, detailing the creation of groups for extrusion.
- Enhanced descriptions of soil layers and bridge materials, including properties and porosity.
- Improved the organization of geometry definitions for clarity.
- Updated load section to reflect the moving load parameters and their application.
- Revised boundary conditions to include fixed base and absorbing boundaries.
- Enhanced solver settings for clarity and updated parameters.
- Added JSON output settings for specific points along the embankment.
- Streamlined the run section to clarify the calculation process.

brunozc committed:
Refactor tutorial documentation for clarity and consistency

- Updated tutorial3.rst to improve formatting and clarity in solver settings and output settings sections.
- Enhanced tutorial4.rst by refining language, correcting grammar, and improving the structure of the train model setup and boundary conditions.
- Revised tutorial5.rst to ensure consistent formatting and clarity in the introduction of necessary packages for model setup.

brunozc committed:
Merge branch 'docs' of github.com:StemVibrations/STEM into docs

brunozc committed:
Update tutorial documentation: fix variable names and adjust mesh size settings

brunozc committed:
Update nightly build workflow: adjust permissions and modify PDF upload process

brunozc committed:
Change uploaded file from report.pdf to benchmark_report.pdf

brunozc committed:
Update file path for benchmark report in workflow

brunozc committed:
Fix formatting in nightly build workflow and disable release notes generation

brunozc committed:
Add bridge animation and update tutorial documentation

- Introduced a new bridge animation GIF to visualize vertical displacement due to a moving load.
- Updated tutorial3.rst to include the new animation.
- Cleaned up code blocks in tutorial2.rst and tutorial4.rst for better readability.

brunozc committed:
Update authors and index documentation: add project description and fix typos

brunozc committed:
Update documentation for visualization and outputs; adjust tutorial formatting

brunozc committed:
Refactor code structure for improved readability and maintainability

brunozc committed:
Refactor moving load halfspace and one-dimensional absorbing boundary scripts; update Rayleigh damping parameters and boundary conditions

- Removed unnecessary roller displacement constraint in `moving_load_halfspace.py`.
- Updated Rayleigh damping parameters in `moving_load_halfspace.py` for improved accuracy.
- Adjusted porosity and Poisson ratio values in `one_dim_abs_boundary.py` to reflect correct material properties.
- Refactored file paths in `runner.py` to use `Path` objects for better compatibility and readability.
- Updated comparison functions to reflect changes in output file names and paths.
- Modified LaTeX documentation for strip load and vibrating dam to correct damping ratio and figure labels.
- Updated SVG and PDF files for vibrating dam mesh to reflect recent changes in dimensions and styles.

brunozc committed:
Add UTF-8 encoding to file open in test_tutorial_docs

Specify UTF-8 encoding when opening the file.

brunozc committed:
Add validation and benchmarking report for STEM model

brunozc committed:
Clarify discussion on accuracy of STEM solution in moving load benchmark; note high frequency content related to domain truncation and boundary conditions.

brunozc committed:
Fix formatting inconsistencies in tutorial documentation

brunozc committed:
Remove redundant text from figure caption in moving load documentation

brunozc committed:
Fix documentation and update analytical solution references; modify SVG and PDF files for improved accuracy and presentation; adjust parameters in comparison script for better modeling fidelity.

aronnoordam commented on docs/api.rst:
```diff
@@ -0,0 +1,16 @@
+each Python module, so every public function appears under the file it lives in
+along with its signature and docstring-derived description.
+
+Explore the package tree below to drill into any module. The generated pages live
```
<3

aronnoordam commented on docs/API_definition.rst:
```diff
@@ -48,7 +48,10 @@ In this example the uvec_data contains the structure that it is used in STEM. Th
-An example of the usage of a UVEC file can be found in the :ref:`tutorial3`.
+An example of the usage of a UVEC file can be found in the following tutorials:
+- :ref:`tutorial4`
+- :ref:`tutorial6`.
```
there is no tutorial6.rst

aronnoordam commented on docs/boundary_conditions.rst:
```diff
@@ -0,0 +1,94 @@
+STEM supports the following types of boundary conditions:
+
+- Displacement constraints (fixed, roller, etc.)
+- Absorbing boundaries (Lysmer-type :cite:`Lysmer_Kuhlemeyer_1969`)
```
also rotation constraints

aronnoordam commented on docs/boundary_conditions.rst:
```diff
@@ -0,0 +1,94 @@
+A value of 1.0 corresponds to impedance-matched (optimal) absorption for normally incident waves,
+while a value of 0.0 disables absorption.
+
+The ``virtual_thickness`` defines a virtual boundary layer used to compute elastic stiffness terms that stabilize
```
defines the thickness of a virtual boundary layer...

brunozc replied:
added:  The greater the virtual thickness, the lower the stiffness at the boundary.

aronnoordam commented on docs/formulation.rst:
```diff
@@ -0,0 +1,326 @@
+:math:`\mathbf{a}`, :math:`\mathbf{v}`, :math:`\mathbf{u}` are, respectively, the acceleration, the velocity and
+the displacement in the nodes.
+
+In STEM is also possible to perform quasi-static analyses, in which inertial and damping effects are neglected.
```
```suggestion
In STEM it is also possible to perform quasi-static analyses, in which inertial and damping effects are neglected.
```

aronnoordam commented on docs/formulation.rst:
```diff
@@ -0,0 +1,326 @@
+STEM support different types of boundary conditions, including:
+
+- Dirichlet boundary conditions (prescribed displacement or velocity)
+- Neumann boundary conditions (force or traction boundary conditions)
```
this is not neuman

aronnoordam commented on docs/formulation.rst:
```diff
@@ -0,0 +1,326 @@
+===================
+STEM supports linear elastic material models.
+Non-linear material models can be implemented by the user and integrated into STEM through the
+UMAT interface (see :doc:`API_definition`).
```
or UDSM

aronnoordam commented on docs/formulation.rst:
```diff
@@ -0,0 +1,326 @@
+
+Linear elastic material
+-----------------------
+The default material model for STEM analysis is the linear elastic material model.
```
there is no default

aronnoordam commented on docs/loads.rst:
```diff
@@ -0,0 +1,209 @@
+
+   mload = MovingLoad(
+       load=[0.0, -1.2e5, 0.0],
+       direction=[1, 0, 0],
```
direction_signs

brunozc committed:
added first draft paper

brunozc committed:
Remove draft PDF workflow and paper markdown file from repository

aronnoordam commented on docs/tutorial2.rst:
```diff
@@ -1,317 +1,387 @@
+Mesh
+----
+The mesh size and element order are defined.
+The element size for the mesh can be defined as a single value, which will applied to the whole model.
```
```suggestion
The element size for the mesh can be defined as a single value, which will be applied to the whole model.
```

aronnoordam commented on docs/tutorial4.rst:
```diff
@@ -1,365 +1,508 @@
\ No newline at end of file
+.. _tutorial4:
+
+Train model (UVEC) on track and embankment in 3D with irregularities
```
and multistage

brunozc committed:
Update mesh.svg dimensions and positioning for improved layout

aronnoordam commented on nightly_build/moving_load_halfspace/moving_load.tex:
```diff
@@ -0,0 +1,85 @@
+for a moving load travelling on an elastic halfspace.
+
+The analytical solution is presented in~\cite{Liao_Teng_Yeh_2005}.
+The analytical solution provides closed-form expressions for the vertical displacement along the surface of the
```
didnt you have to check below the surface of the halfspace?

aronnoordam commented on nightly_build/moving_load_halfspace/moving_load.tex:
```diff
@@ -0,0 +1,85 @@
+
+The analytical solution is presented in~\cite{Liao_Teng_Yeh_2005}.
+The analytical solution provides closed-form expressions for the vertical displacement along the surface of the
+halfspace, enabling a direct time-history comparison against the numerical model.
```
doesnt the analytical solution provide expressions below the surface?

brunozc replied:
yes

aronnoordam commented on nightly_build/moving_load_halfspace/moving_load.tex:
```diff
@@ -0,0 +1,85 @@
+	\item Young's modulus: \qty{50}{\mega\pascal},
+	\item Poisson ratio: \qty{0.3}{},
+	\item Solid density: \qty{2700}{\kilogram\per\meter\cubed},
+	\item Porosity: 0.3.
```
maybe combine these to a bulk density?

aronnoordam commented on nightly_build/stem_runners/compare_analytical.py:
```diff
@@ -0,0 +1,811 @@
+    plt.close()
+
+
+def compare_strip_load_3D(path_model, output_file):
```
this can be removed righT? since it is there in compare_strip_load

aronnoordam commented on nightly_build/moving_load_halfspace/moving_load.tex:
```diff
@@ -0,0 +1,85 @@
+
+\begin{figure}
+	\centering
+	\includegraphics[width=0.75\textwidth]{moving_load_halfspace/mesh.pdf}
```
the figure is really small in the docs

aronnoordam reviewed (commented):
few remarks

aronnoordam reviewed (commented):
few remarks

brunozc commented on docs/boundary_conditions.rst:
```diff
@@ -0,0 +1,94 @@
+
+   # Apply to surface IDs (dimension=2 for surfaces in 3D)
+   model.add_boundary_condition_by_geometry_ids(2, [1], fixed, "base_fixed")
+   model.add_boundary_condition_by_geometry_ids(2, [2, 4, 5, 6, 7, 10, 11, 12, 15, 16, 17], roller, "sides_roller")
```
changed

brunozc commented on docs/solver_settings.rst:
```diff
@@ -0,0 +1,148 @@
+       f2 = 80  # Hz
+       d2 = 0.02  # 2% damping at f2
+
+       alpha, beta = damping_Rayleigh(f1, d1, f2, d2)
```
changed

brunozc commented on docs/solver_settings.rst:
added

Linear solver type
------------------
STEM provides several linear solvers:
STEM has several linear solvers available, including:

- LU decomposition (direct)
- Conjugate Gradient (iterative)
- Algebraic Multigrid (iterative)

LU decomposition is best suited for small problems.
For large problems, the iterative solvers are recommended, as they scale better and can use
preconditioners to improve convergence.
STEM uses the Jacobi preconditioner by default, a simple diagonal preconditioner suitable for symmetric
positive definite systems.

brunozc commented on docs/track.rst:
added: 

To reduce the size of the soil domain, the railway track can be generated partially outside the model geometry
by extending it beyond the domain boundaries.
The extended part of the track is supported by a one-dimensional spring-damper system,
which represents the soil behavior using 1D elements.

An extended straight track can be generated as follows:

.. code-block:: python
   
   direction_vector = [0, 0, 1]
   n_sleepers = 0.6
   number_of_sleepers = 101
   sleeper_spacing = 0.6
   rail_pad_thickness = 0.025
   length_soil_equivalent_element = 5
   name = "extended_track"
    rail_pad_thickness = 0.025

    model.generate_extended_straight_track(n_sleepers,
                                           number_of_sleepers,
                                           rail_parameters,
                                           sleeper_parameters,
                                           rail_pad_parameters,
                                           rail_pad_thickness,
                                           origin_point,
                                           soil_equivalent_parameters,
                                           length_soil_equivalent_element,
                                           direction_vector,
                                           name)

The figure below illustrates an example of a railway track generated with an extended track section.

.. figure:: _static/extended_track.png
   :align: center
   :alt: Extended track
   :width: 400

brunozc commented on docs/tutorial2.rst:
```diff
@@ -1,317 +1,387 @@
+    roller_displacement_parameters = DisplacementConstraint(is_fixed=[True, False, True], value=[0, 0, 0])
+
+    model.add_boundary_condition_by_geometry_ids(2, [1], no_displacement_parameters, "base_fixed")
+    model.add_boundary_condition_by_geometry_ids(2, [2, 4, 5, 6, 7, 10, 11, 12, 16, 17, 18],
```
yes!

brunozc commented on docs/tutorial4.rst:
```diff
@@ -1,365 +1,508 @@
+
+The static initialisation of the train is done by setting the `static_initialisation` parameter to True in the
+`uvec_parameters` dictionary, and the train velocity is set to zero, so that the train does not move
+during the static initialisation stage.
```
"It should be noted that the `uvec_load` is added to a `line model part`, and not to a specific coordinate point."

brunozc reviewed (commented):
done

brunozc committed:
changes for review

aronnoordam commented on docs/track.rst:
```diff
@@ -147,6 +147,42 @@ distance between the middle of the sleeper and the rail as defined above):
+   name = "extended_track"
+    rail_pad_thickness = 0.025
+
+    model.generate_extended_straight_track(n_sleepers,
```
i would also add an instance of soil_equivalent_parameters

brunozc committed:
Merge branch 'main' into docs

brunozc committed:
review comments

brunozc committed:
Update benchmark report to dynamically fetch STEM version and adjust figure widths in vibrating dam documentation

- Modified benchmark_report.tex to read the STEM version from the Python package's version file.
- Commented out the draft watermark settings in benchmark_report.tex.
- Increased the width of subfigures in vibrating_dam.tex from 0.49 to 0.8 for better visibility.
- Updated the vibrating_dam_mesh_3d.pdf file to reflect changes in the SVG source.
- Made adjustments to the vibrating_dam_mesh_3d.svg file, including removing unnecessary gradients and updating element IDs for clarity.

brunozc committed:
Enhance benchmark report formatting by introducing a new command for consistent font styling of version and commit information

brunozc committed:
Merge branch 'main' into docs

brunozc committed:
Update nightly benchmark report schedule to run at 10 AM
